src/path_ordering.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2023 UltiMaker
+// CuraEngine is released under the terms of the AGPLv3 or higher.
+
+#include "path_ordering.h" //The definitions we're implementing here.
+
+#include "WallToolPaths.h"
+#include "sliceDataStorage.h" //For SliceLayerPart.
+
+namespace cura
+{
+
+template<>
+const PointsSet& PathOrdering<const Polygon*>::getVertexData()
+{
+    return *vertices_;
+}
+
+template<>
+const PointsSet& PathOrdering<Polygon*>::getVertexData()
+{
+    return *vertices_;
+}
+
+template<>
+const PointsSet& PathOrdering<const OpenPolyline*>::getVertexData()
+{
+    return *vertices_;
+}
+
+template<>
+const PointsSet& PathOrdering<OpenPolyline*>::getVertexData()
+{
+    return *vertices_;
+}
+
+template<>
+const PointsSet& PathOrdering<ClosedPolyline*>::getVertexData()
+{
+    return *vertices_;
+}
+
+template<>
+const PointsSet& PathOrdering<Polyline const*>::getVertexData()
+{
+    return *vertices_;
+}
```

Comment:
Can't this be a default templated function? (That'll need to be in the header of course.)

Reply:
I tried to remove template functions from headers as much as possible because that forces you to include a lot of headers. But that could probably be reduced somehow (like using template function instanciation, which I did a lot in some other files)

Reply:
Template instanciation did work, nice :smile: